File: lecture008/multMatrix.java
import java.util.Scanner;
public class multMatrix{
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args){
        System.out.println("enter rows and col value : ");
        int n = scn.nextInt();
        int m = scn.nextInt();
        int[][] arr = new int[n][m];
        System.out.println("enter 1st array : ");
        input(arr);
        System.out.println("enter rows and col : ");
        int p=scn.nextInt();
        int q=scn.nextInt();
        int [][] ar= new int[p][q];
        System.out.println("enter 2nd array : ");
        input(ar);
        multi(arr , n , m , ar , p , q);
        //print(arr);
    }
    public static void input(int [][] a){
        for(int i=0 ; i<a.length ; i++){
            for(int j=0 ; j<a[0].length ; j++){
                a[i][j]=scn.nextInt();
            }
        }
       
    }
    public static void multi(int [][] a, int n , int m , int [][] b , int p ,int q){
        // int n=a.length;
        // int m=a[0].length;
        // int p=b.length;
        // int q=b[0].length;
        if(m==p){
            int[][] res=new int[n][q];
            for(int i=0 ; i<a.length ; i++){
                for(int j=0 ; j<b[0].length ; j++){
                   // res[i][j]=0;
                    for(int k=0 ; k<m ; k++){
                        res[i][j]+=a[i][k]*b[k][j];
                    }
                }
            }
            System.out.print("multiplied matrix is ");
            print(res);
        }
        else{
            System.out.print("matrix cannot be multiplied");
        }
        
        
    }
    public static void print(int [][] a){
        for(int i=0 ; i<a.length ; i++){
            System.out.println();
            for(int j=0 ; j<a[0].length ; j++){
                System.out.print(a[i][j]+ " ");
            }
        }
        
    }

}

File: lecture008/multMatrix.cpp
#include<iostream>
#include<vector>
using namespace std;
void input(vector<vector<int>> &a){
    for(int i=0 ; i<a.size() ; i++){
        for(int j=0 ; j<a[0].size() ; j++){
            cin>>a[i][j];
        }
    }

}
void print(vector<vector<int>> &a){
    for(int i=0 ; i<a.size() ; i++){
        for(int j=0 ; j<a[0].size() ; j++){
            cout<<a[i][j]<<" ";
        }
    }

}
void multiplication(vector<vector<int>> &a , int m , int n , vector<vector<int>> &b , int p , int q){
    vector<vector<int>> res(m ,vector<int> (q,0));
    if(m==p){
        for(int i=0 ; i<a.size() ; i++){
        for(int j=0 ; j<b[0].size() ; j++){
            //res[i][j]=0;
            for(int k=0 ; k<m ; k++){
                res[i][j]+=a[i][j]*b[i][j];
            }

        }
    }
    cout<<"multiplied matrix is "<<endl;
    print(res);
    }
    else{
        cout<<"cannot be multiplied";
    }
}
int main(int args , char**argv){
    int n,m,p,q;
    cout<<"enter row and col of 1st array : "<<endl;
    cin>>m>>n;
    cout<<"enter row and col of 2nd array : "<<endl;
    cin>>p>>q;
    vector<vector<int>> arr(m , vector<int>(n,0)); 
    cout<<"enter 1st array "<<endl;
    input(arr);
    vector<vector<int>> ar(p , vector<int>(q,0));
    cout<<"enter 2nd array "<<endl;
    input(ar);
    multiplication(arr , m , n , ar , p , q);
    //print(arr);
}